inc/template.php

```diff
@@ -21,19 +21,21 @@
  * @return int[] Array of user IDs.
  */
 function get_author_ids( WP_Post $post ) : array {
-	if ( ! is_post_type_supported( $post->post_type ) ) {
-		if ( post_type_supports( $post->post_type, 'author' ) ) {
-			return [ intval( $post->post_author ) ];
+	$authors = wp_get_post_terms( $post->ID, TAXONOMY );
+	// If $authors is an error or empty.
+	if ( is_wp_error( $authors ) || empty( $authors ) ) {
+		// Check if the post_type supports Authorship.
+		if ( is_post_type_supported( $post->post_type ) ) {
```

Comment:
This changes the existing logic but it’s not clear why. For any post type that doesn’t support authorship the function will now always return an empty array

Reply:
I'll test some more and see if I can tweak this further

Reply:
I'm not sure why this change was needed at all to be honest, what lead you to altering it?

Reply:
The function `get_author_ids` currently checks if a post type is not supported and then followed by doing a `wp_get_post_terms` which if the user is a super admin it returns an empty array.

So my thinking was to perform the check first of `wp_get_post_terms` if it return an error or was empty then it would return the post author details. Looking at it now it could probably do with dropping the `is_post_type_supported` check too.

Reply:
I think it can just be put back how it was - the problem was fixed just adding the `blog_id => 0` argument right?

The change you have here will now cause a DB query even if the post type does not support multiple authors. Sometimes condensing logic like this doesn't help with performance or readability.

It's always good to do one thing at a time, ideally starting with conditionals that are the least effort to process, e.g. not checking the db.

Reply:
`blog_id => 0` allows us to get the user who isn't assigned to the existing site when we have an ID. However with the current code in place for `get_author_ids` it returns an empty array still which will still show the reported issue.

So this change will ensure an author ID is returned.

Reply:
Could the problem be that when saving the author terms they aren't being stored if the user doesn't have a role on the current site?

This is still hiding the real issue I think - `wp_get_post_terms( $post->ID, TAXONOMY );` should be returning IDs even for users that don't strictly have a role on the site.